node/src/components/fetcher/tests.rs

```diff
@@ -89,9 +89,20 @@ reactor!(Reactor {
                 .chainspec()
                 .protocol_config
                 .last_emergency_restart,
+            chainspec_loader
+                .chainspec()
+                .protocol_config
+                .merkle_tree_hash_activation,
```

Comment:
Yes, that's a good idea. We should even use this approach in `ContractRuntime::new()`, but as agreed on a call, we should attack it in the separate PR mainly because it will require substantial changes to `retrieve-state` tool, for example 
[here](https://github.com/casper-network/casper-node/blob/release-1.4.3/utils/retrieve-state/src/storage.rs#L128-L139) we'd need to create an entire, dummy chainspec to pass it to the constructor.

It'll work nicely after the suggestion from [this comment](https://github.com/casper-network/casper-node/pull/2531#discussion_r778709175) is implemented, though.

I will leave the code as-is and request a `technical-debt` ticket.